stdlib/public/core/AnyHashable.swift

```diff
@@ -120,7 +120,7 @@ internal struct _ConcreteHashableBox<Base : Hashable> : _AnyHashableBox {
 ///         AnyHashable(Set(["a", "b"])): "a set of strings"
 ///     ]
 ///     print(descriptions[AnyHashable(42)]!)      // prints "an Int"
-///     print(descriptions[AnyHashable(43)])       // prints "nil"
+///     print(descriptions[AnyHashable(45)])       // prints "nil"
```

Comment:
It might be more illustrative to keep the `descriptions[AnyHashable(43)]` example here, and update the 'prints "nil"' comment to show the expected behavior.